unity-renderer/Assets/Scripts/Tests/TestUtils.cs

```diff
@@ -858,9 +858,9 @@ public static IEnumerator TestShapeCollision(BaseShape shapeComponent, BaseShape
             yield return shapeComponent.routine;
 
             // check colliders correct behaviour
-            for (int i = 0; i < entity.meshesInfo.colliders.Count; i++)
+            foreach (Collider collider in entity.meshesInfo.colliders)
```

Comment:
i would consider the fact that `foreach` introduces a GC issue compared to standard `for`.. if this method is not frequently called, then ignore the comment

Reply:
The GC problems that the older MONO compiler used in Unity had, have been solved since long time ago, some sources [here](https://answers.unity.com/questions/1359179/does-unity-fixed-foreach-problem-gc-boxing.html) and [here](https://www.jacksondunstan.com/articles/3805). 

Regardless of that, since that collection traversing is happening in a Test environment I think we'd be safe anyway